app/src/main/java/org/metabrainz/mobile/repository/SearchRepository.java

```diff
@@ -0,0 +1,205 @@
+package org.metabrainz.mobile.repository;
+
+import androidx.lifecycle.MutableLiveData;
+
+import org.metabrainz.mobile.api.data.search.entity.Artist;
+import org.metabrainz.mobile.api.data.search.entity.Event;
+import org.metabrainz.mobile.api.data.search.entity.Instrument;
+import org.metabrainz.mobile.api.data.search.entity.Label;
+import org.metabrainz.mobile.api.data.search.entity.Recording;
+import org.metabrainz.mobile.api.data.search.entity.Release;
+import org.metabrainz.mobile.api.data.search.entity.ReleaseGroup;
+import org.metabrainz.mobile.api.data.search.response.ArtistSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.EventSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.InstrumentSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.LabelSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.RecordingSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.ReleaseGroupSearchResponse;
+import org.metabrainz.mobile.api.data.search.response.ReleaseSearchResponse;
+import org.metabrainz.mobile.api.webservice.Constants;
+import org.metabrainz.mobile.api.webservice.MusicBrainzService;
+import org.metabrainz.mobile.api.webservice.MusicBrainzServiceGenerator;
+import org.metabrainz.mobile.util.Log;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class SearchRepository {
```

Comment:
Singleton Repository class to query MB WS2 and post search results.

Reply:
Do you know Kotlin? In case you know, how would you see that for the new files make them in Kotlin. Both can co-exist and I think that it's a step forward in Android features.

However, this is just a proposal that depends on your expertise on this new language. Let me know your thoughts, please :)

Reply:
I know Kotlin a bit. But I am not an expert so I don't know the side effects. Therefore, I am currently keeping everything in Java. I will continue to study Kotlin. At a later stage, we might port to Kotlin.